packages/permission-controller/src/PermissionController.ts

```diff
@@ -1181,6 +1191,67 @@ export class PermissionController<
     this.setCaveat(origin, target, caveatType, caveatValue);
   }
 
+  /**
+   * Updates caveat value across all subjects' permissions that have the specified
+   * target and caveat type.
+   *
+   * @template TargetName - The permission target name. Should be inferred.
+   * @template CaveatType - The valid caveat type for the permission. Should
+   * be inferred.
+   * @param target - The target name of the permission.
+   * @param caveatType - The type of the caveat to update.
+   * @param caveatValue - The new value of the caveat.
+   */
+  updateCaveatForAllSubjects<
+    TargetName extends ExtractPermission<
+      ControllerPermissionSpecification,
+      ControllerCaveatSpecification
+    >['parentCapability'],
+    CaveatType extends ExtractAllowedCaveatTypes<ControllerPermissionSpecification>,
+    CaveatValue extends ExtractCaveatValue<
+      ControllerCaveatSpecification,
+      CaveatType
+    >,
+  >(
+    target: TargetName,
+    caveatType: CaveatType,
+    caveatValue: CaveatValue,
+  ): void {
+    const specification = this.getCaveatSpecification(caveatType);
+    const caveat = { type: caveatType, value: caveatValue };
```

Comment:
Is it up to the callee to merge the existing caveat with the new one or should this function support that? Some caveat values may be mergeable, e.g. arrays or objects.

Reply:
Up to the callee to merge the existing caveat. I thought about this point a little and I didn't want to introduce any opinion as to the shape of the caveat value itself since its type is `Json`

Reply:
Fair point